server/vcs/provider/github/file_fetcher.go

```diff
@@ -52,3 +67,40 @@ func (r *RemoteFileFetcher) GetModifiedFilesFromCommit(ctx context.Context, repo
 	}
 	return files, nil
 }
+
+// GetModifiedFilesFromPR returns the names of files that were modified in the pull request
+// relative to the repo root, e.g. parent/child/file.txt.
+func (r *RemoteFileFetcher) getModifiedFilesFromPR(ctx context.Context, repo models.Repo, pullNum int, installationToken int64) ([]string, error) {
+	var files []string
+	nextPage := 0
+	for {
+		opts := gh.ListOptions{
+			PerPage: 300,
+		}
+		if nextPage != 0 {
+			opts.Page = nextPage
+		}
+		client, err := r.ClientCreator.NewInstallationClient(installationToken)
```

Comment:
any reason why we're creating a client inside the for loop for every iteration?

Reply:
+1 let's move this outside the loop.